pipeline_tools/create_analysis_metadata.py

```diff
@@ -0,0 +1,502 @@
+                                                        schema_version=analysis_process_schema_version),
+        'schema_type'        : SCHEMA_TYPE,
+        'process_core'       : get_analysis_process_core(analysis_workflow_id=analysis_id),
+        'process_type'       : get_analysis_process_type(),
+        'timestamp_start_utc': workflow_metadata.get('start'),
+        'timestamp_stop_utc' : workflow_metadata.get('end'),
+        'input_bundles'      : input_bundles_string.split('.'),
+        'reference_bundle'   : reference_bundle,
+        'tasks'              : workflow_tasks,
+        'inputs'             : inputs,
+        'outputs'            : get_outputs(output_url_to_md5=output_url_to_md5,
+                                           extension_to_format=extension_to_format,
+                                           schema_url=raw_schema_url,
+                                           analysis_file_version=analysis_file_version),
+        'analysis_run_type'  : run_type,
+    }
+    return analysis_process
+
+
+def create_analysis_protocol(raw_schema_url, analysis_protocol_schema_version, pipeline_version, method):
+    """Collect and create the information about the analysis protocol for submission to Ingest service.
+
+    Based the design of this HCA metadata type, one analysis_protocol will be shared by every analysis run for a given
+    pipeline, although the same content will be submitted to the protocols endpoint for each run. Besides, any changes
+    to the pipeline version will change the content of the standard analysis_protocol. This function strictly follows
+    the metadata schema defined at: https://github.com/HumanCellAtlas/metadata-schema
+
+    TODO: Implement the dataclass in "https://github.com/HumanCellAtlas/metadata-api/blob/1b7192cecbef43b5befecc4153bf
+    2e2f4db5bb16/src/humancellatlas/data/metadata/__init__.py#L339" so we can use the `metadata-api` directly to create
+    the analysis_protocol metadata file.
+
+    Args:
+        pipeline_version (str): The version of the pipeline, usually provided by the label of the adapter workflow
+                                around the analysis workflow.
+        raw_schema_url (str): URL prefix for retrieving HCA metadata schemas.
+        analysis_protocol_schema_version (str): Version of the metadata schema that the analysis_protocol conforms to.
+        method (str): The name of the analysis workflow, e.g. "SmartSeq2SingleCell"
+
+    Returns:
+        analysis_protocol (dict): A dict representing the analysis_protocol json file to be submitted.
+    """
+    SCHEMA_TYPE = 'protocol'
+
+    analysis_protocol = {
+        'describedBy'         : get_analysis_describedBy(schema_url=raw_schema_url,
+                                                         schema_type=SCHEMA_TYPE,
+                                                         schema_version=analysis_protocol_schema_version),
+        'schema_type'         : SCHEMA_TYPE,
```

Comment:
To all this type of comments: the extra spaces here are intentional 🌞 Although they are violating the PEP 8, I found a lot of repos are adopting this style, and it can be turned on in PyCharm -> Tools -> Editor -> Code Style -> Python -> Other -> `Align on colon` Personally I feel it makes code more human readable, consider:

```python
'schema_type'        : SCHEMA_TYPE,
'process_core'       : get_analysis_process_core(analysis_workflow_id=analysis_id),
'process_type'       : get_analysis_process_type(),
'timestamp_start_utc': workflow_metadata.get('start'),
'timestamp_stop_utc' : workflow_metadata.get('end'),
'input_bundles'      : input_bundles_string.split(','),
'reference_bundle'   : reference_bundle,
'tasks'              : workflow_tasks,
'inputs'             : inputs,
```
compare to:

```python
'schema_type' : SCHEMA_TYPE,
'process_core': get_analysis_process_core(analysis_workflow_id=analysis_id),
'process_type': get_analysis_process_type(),
'timestamp_start_utc': workflow_metadata.get('start'),
'timestamp_stop_utc' : workflow_metadata.get('end'),
'input_bundles': input_bundles_string.split(','),
'reference_bundle': reference_bundle,
'tasks': workflow_tasks,
'inputs': inputs,
```

But, this is ONLY my personal preferences, I'm happy to change it if we'd reach some agreements on code style (or just using a linter like we did in jmui)